Sources/ArgumentParser/Parsing/ArgumentSet.swift

```diff
@@ -171,23 +171,44 @@ extension ArgumentSet {
   }
   
   /// Creates an argument set for a pair of inverted Boolean flags.
-  static func flag(key: InputKey, name: NameSpecification, default initialValue: Bool?, inversion: FlagInversion, help: ArgumentHelp?) -> ArgumentSet {
+  static func flag(key: InputKey, name: NameSpecification, default initialValue: Bool?, inversion: FlagInversion, exclusivity: FlagExclusivity, help: ArgumentHelp?) -> ArgumentSet {
     // The flag is required if initialValue is `nil`, otherwise it's optional
     let helpOptions: ArgumentDefinition.Help.Options = initialValue != nil ? .isOptional : []
     
-    let help = ArgumentDefinition.Help(options: helpOptions, help: help, defaultValue: initialValue.map(String.init), key: key)
     let (enableNames, disableNames) = inversion.enableDisableNamePair(for: key, name: name)
+
+    var hasUpdated = false
+    func boolUpdate(origin: InputOrigin, name: Name?, values: inout ParsedValues) throws {
+      let boolValue = name.map(enableNames.contains) ?? false
+      let previous = values.element(forKey: key)
+      switch (hasUpdated, previous, exclusivity) {
+      case (true, let p?, .exclusive):
+        // This value has already been set.
+        throw ParserError.duplicateExclusiveValues(previous: p.inputOrigin, duplicate: origin, originalInput: values.originalInput)
+      case (false, _, _), (_, _, .chooseLast):
+        values.set(boolValue, forKey: key, inputOrigin: origin)
+        hasUpdated = true
+      default:
+        break
+      }
+    }
```

Comment:
This is more or less repeated code from `Flag.swift` lines 219-230. I can see that it will be required another time, so it should probably be refactored to a function.